sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsPartitions.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connector.catalog;
+
+import java.util.Map;
+
+import org.apache.spark.annotation.Experimental;
+
+/**
+ * Catalog methods for working with Partitions.
+ */
+@Experimental
+public interface SupportsPartitions extends TableCatalog {
+
+    /**
+     * Create partitions in an existing table, assuming it exists.
+     *
+     * @param ident a table identifier
+     * @param partitions transforms to use for partitioning data in the table
+     * @param ignoreIfExists
+     */
+    void createPartitions(
+            Identifier ident,
+            TablePartition[] partitions,
+            Boolean ignoreIfExists);
+
+    /**
+     * Drop partitions from a table, assuming they exist.
+     *
+     * @param ident a table identifier
+     * @param partitions a list of string map for existing partitions
+     * @param ignoreIfNotExists
+     */
+    void dropPartitions(
```

Comment:
In other places in the v2 API, we leave `ifNotExists` to Spark so that we don't need to pass it. Instead of passing an extra parameter, the method returns a boolean to indicate whether the partition was dropped or if no action was taken. Either way, the partition should not exist after the method call so it is idempotent.

Then, Spark decides whether to throw an exception because the partition did not exist. We prefer that pattern of doing more in Spark to make behavior standard across sources and to make the requirements as simple as we can.

Reply:
Good point, I will change it.

Reply:
This can be renamed to `boolean dropPartition(identifier)`